_papers/inspToMd.py

```diff
@@ -1,74 +1,114 @@
+    returned by `sxs.utilities.inspire.query`) and produce a markdown string to
+    be written to a file. The optional arguments `used_spec` and `used_spectre`
+    will set the corresponding values in the markdown."""
+
+    iid = resp['id']
+    md = resp['metadata']
+    if (len(md['texkeys']) > 1):
+        warn(f"More than 1 texkeys in {iid}; using first.")
+    texkey = md['texkeys'][0]
+    if (len(md['titles']) > 1):
+        warn(f"More than 1 titles in {iid}; using first.")
+    title = md['titles'][0]['title']
+    authors = [a['full_name'] for a in md['authors']]
+    if len(authors) == 1:
+        authors_str = f" \"{authors[0]}\""
+    else:
+        authors_str = "\n" + "\n".join([f"  - \"{a}\"" for a in authors])
+    if 'publication_info' in md:
+        pub_info = md['publication_info']
+        if (len(pub_info) > 1):
+            warn(f"More than 1 publication_info in {iid}; using first.")
+        pub_info = pub_info[0]
+        jref_str = f" \"{pub_info.get('journal_title','')} " + \
+            f"{pub_info.get('journal_volume','')}, " + \
+            f"{pub_info.get('artid','')} ({pub_info.get('year','')})\""
+    else:
+        jref_str = ""
+    date = md['earliest_date']
+    if 'arxiv_eprints' in md:
+        if (len(md['arxiv_eprints']) > 1):
+            warn(f"More than 1 arxiv #s in {iid}; using first.")
+        arxiv_str = f" \"{md['arxiv_eprints'][0]['value']}\""
+    else:
+        arxiv_str = ""
+    if 'dois' in md:
+        if (len(md['dois']) > 1):
+            warn(f"More than 1 dois in {iid}; using first.")
+        doi_str = f" \"{md['dois'][0]['value']}\""
+    else:
+        doi_str = ""
+    if (len(md['abstracts']) > 1):
+        warn(f"More than 1 abstracts in {iid}; using first.")
+    abstract_str = md['abstracts'][0]['value']
+    abstract_str = fill(abstract_str,
+                        initial_indent='  ',
+                        subsequent_indent='  ',
+                        break_long_words=False)
+    if used_spec is None:
+        used_spec = False
+    used_spec_str = " true" if used_spec else ""
+    if used_spectre is None:
+        used_spectre = False
+    used_spectre_str = " true" if used_spectre else ""
+
+    return f"""---
 title: "{title}"
 authors:{authors_str}
 jref:{jref_str}
 doi:{doi_str}
 date: {date}
 arxiv:{arxiv_str}
 insp_recid: {iid}
-used_spec:
-used_spectre:
+used_spec:{used_spec_str}
+used_spectre:{used_spectre_str}
 abstract: |
 {abstract_str}
 ---
-""")
+"""
+
+def write_insp_resp_to_md(responses):
+    """Take a JSON response from INSPIRE (e.g. return from
+    `sxs.utilities.inspire.query`) and write it to a bunch of .md files"""
+    for resp in responses:
+        md = resp['metadata']
+        # If there's no texkey, we don't know what to do
+        if 'texkeys' in md and (len(md['texkeys']) > 0):
+            texkey = md['texkeys'][0]
+        else:
+            warn(f"Didn't find a texkey in {iid}; skipping!")
```

Comment:
I *think* every INSPIRE record should have a texkey... before this change, the script would fail if it got a response without a texkey. Now it won't fail, but it will basically be silent (unless somebody looks at the gh action log)